nixos/modules/services/backup/bacula.nix

```diff
@@ -9,6 +9,21 @@ with lib;
 let
   libDir = "/var/lib/bacula";
 
+  yes_no = bool: if bool then "yes" else "no";
+  tls_conf = tls_cfg: optionalString tls_cfg.enable (
+    concatStringsSep
+      "\n"
+      (
+      [''TLS Enable = yes;'']
+      ++ optional (tls_cfg.require != null) ''TLS Require = ${yes_no tls_cfg.require};''
+      ++ optional (tls_cfg.certificate != null) ''TLS Certificate = "${tls_cfg.certificate}";''
+      ++ [''TLS Key = "${tls_cfg.key}";'']
+      ++ optional (tls_cfg.verifyPeer != null) ''TLS Verify Peer = ${yes_no tls_cfg.verifyPeer};''
+      ++ optional (tls_cfg.allowedCN != { }) ''TLS Allowed CN = ${concatStringsSep " " (attrNames tls_cfg.allowedCN)};''
+      ++ optional (tls_cfg.caCertificateFile != null) ''TLS CA Certificate File = ${tls_cfg.caCertificateFile};''
```

Comment:
```suggestion
      ++ optional (tls_cfg.allowedCN != { }) ''TLS Allowed CN = ${concatStringsSep " " tls_cfg.allowedCN};''
      ++ [''TLS CA Certificate File = "${tls_cfg.caCertificateFile}";'']
```

Reply:
As you can see at [Bacula Docs](https://www.bacula.org/11.0.x-manuals/en/main/Configuring_Director.html#SECTION0021200000000000000000) under `TLS Allowed CN = ` you are not supposed to add quotes to `CA Certificate File`.

Reply:
> As you can see at the [Bacula Docs](https://www.bacula.org/11.0.x-manuals/en/main/Configuring_Director.html#SECTION0021200000000000000000) under `TLS Allowed CN = ` you are not supposed to add quotes to `CA Certificate File`.

Correct me if I'm wrong, but while the example doesn't use quotes, `TLS CA Certificate File` accepts datatype `<Filename>` just like `TLS Key` and `TLS Certificate`. So shouldn't we treat these three options in the same way (meaning either quotes for all three of them or for non of them)?

Reply:
Yes you are Correct. But I am honestly not completely sure  witch is the better way. Since on the one hand the Documentation does not use Quotes and therefor it is the recommended style. On the other hand I am not sure about special characters like space in filenames.

Reply:
Just looked at the Documentation for the [Data Types](https://www.bacula.org/11.0.x-manuals/en/main/Customizing_Configuration_F.html#SECTION0020240000000000000000) 

> A directory is either a quoted or non-quoted string.

So lets use quoted strings just for easier handling of special characters.